site/src/pages/tutorials/thrift/blog/implement-create.mdx

```diff
@@ -0,0 +1,160 @@
+  ...
+  @Override
+  public void createBlogPost(CreateBlogPostRequest request, AsyncMethodCallback<BlogPost> resultHandler)
+          throws TException {
+    final int id = idGenerator.getAndIncrement();
+    final Instant now = Instant.now();
+    final BlogPost blogPost = new BlogPost()
+            .setId(id)
+            .setTitle(request.getTitle())
+            .setContent(request.getContent())
+            .setModifiedAt(now.toEpochMilli())
+            .setCreatedAt(now.toEpochMilli());
+  }
+  ```
+2. In the `createBlogPost()` method, save the post information in the `blogPosts` map and return the information of the created blog post to the `resultHandler`.
+  ```java filename=BlogServiceImpl.java
+  @Override
+  public void createBlogPost(CreateBlogPostRequest request, AsyncMethodCallback<BlogPost> resultHandler)
+          throws TException {
+    ...
+    blogPosts.put(id, blogPost);
+    final BlogPost stored = blogPost;
+    resultHandler.onComplete(stored);
+  }
+  ```
+
+## 2. Implement client-side
+
+In the `BlogClient` class, add a method to send a request to create a blog post.
+
+```java filename=BlogClient.java
+import example.armeria.blog.thrift.BlogPost;
+import example.armeria.blog.thrift.CreateBlogPostRequest;
+...
+BlogPost createBlogPost(String title, String content) throws TException {
+  final CreateBlogPostRequest request =
+          new CreateBlogPostRequest().setTitle(title)
+                                     .setContent(content);
+  return blogService.createBlogPost(request);
+}
+```
+
+## 3. Set up for testing
+
+Now, let's get started with the test code so that we can verify what we implement along the way.
+Note the code is minimal to keep things simple.
+
+1. Create a file, `BlogServiceTest.java`, and declare the test class.
+  You can see the full version of the file [here](https://github.com/line/armeria-examples/blob/main/tutorials/thrift/src/test/java/example/armeria/server/blog/thrift/BlogServiceTest.java).
+  ```java filename=BlogServiceTest.java
+  package example.armeria.server.blog.thrift;
+
+  class BlogServiceTest {}
+  ```
+2. Register a <type://ServerExtension> to set up a server instance for testing.
+  Note that the service instance is added to the configuration.
+  ```java filename=BlogServiceTest.java
+  import org.junit.jupiter.api.extension.RegisterExtension;
+
+  import com.linecorp.armeria.server.ServerBuilder;
+  import com.linecorp.armeria.server.thrift.THttpService;
+  import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+  class BlogServiceTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+      @Override
+      protected void configure(ServerBuilder sb) throws Exception {
+        sb.service("/thrift", THttpService.builder()
+                                          .addService(new BlogServiceImpl()) // Add the service to the configuration
+                                          .build());
+      }
```

Comment:
We might want to reindent this so it is not too wide:
```suggestion
      protected void configure(ServerBuilder sb) throws Exception {
        sb.service("/thrift",
                   THttpService.builder()
                               // Add the service to the configuration
                               .addService(new BlogServiceImpl())
                               .build());
      }
```
We could even break right before `.builder()` to make it even narrower.